BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
++import net.kyori.adventure.text.Component; // Waterfall - we're going on an adventure!
++import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+ import net.md_5.bungee.api.Callback;
+ import net.md_5.bungee.api.CommandSender;
+ import net.md_5.bungee.api.ProxyServer;
+@@ -46,8 +48,11 @@ public class BungeeServerInfo implements ServerInfo
+     @Getter
+     private final SocketAddress socketAddress;
+     private final Collection<ProxiedPlayer> players = new ArrayList<>();
+-    @Getter
+-    private final String motd;
++    // Waterfall start - we're going on an adventure!
++//    @Getter
++//    private final String motd;
++    private final Component motd;
++    // Waterfall end
+     @Getter
+     private final boolean restricted;
+     @Getter
+@@ -78,6 +83,18 @@ public class BungeeServerInfo implements ServerInfo
+         return "bungeecord.server." + name;
+     }
+ 
++    // Waterfall start - we're going on an adventure!
++    @Override
++    public String getMotd() {
++        return LegacyComponentSerializer.legacySection().serialize(motd);
++    }
++
++    @Override
++    public Component motd() {
++        return motd;
++    }
++    // Waterfall end
++
+     @Override
+     public boolean canAccess(CommandSender player)
+     {
+diff --git a/proxy/src/main/java/net/md_5/bungee/ServerConnection.java b/proxy/src/main/java/net/md_5/bungee/ServerConnection.java
+index 5e452780..5526d3d6 100644
+--- a/proxy/src/main/java/net/md_5/bungee/ServerConnection.java
++++ b/proxy/src/main/java/net/md_5/bungee/ServerConnection.java
+@@ -9,6 +9,7 @@ import lombok.Data;
+ import lombok.Getter;
+ import lombok.RequiredArgsConstructor;
+ import lombok.Setter;
++import net.kyori.adventure.text.Component; // Waterfall start - we're going on an adventure!
+ import net.md_5.bungee.api.chat.BaseComponent;
+ import net.md_5.bungee.api.connection.Server;
+ import net.md_5.bungee.netty.ChannelWrapper;
+@@ -60,6 +61,14 @@ public class ServerConnection implements Server
+         ch.close();
+     }
+ 
++    // Waterfall - we're going on an adventure!
++    @Override
++    public void disconnect(Component reason) {
++        // we ignore the reason rather than throwing like the above does as this should fail silently
++        ch.close();
++    }
++    // Waterfall end
++
+     @Override
+     public void disconnect(BaseComponent reason)
+     {
+diff --git a/proxy/src/main/java/net/md_5/bungee/UserConnection.java b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
+index 7ec119ea..c3b5afea 100644
+--- a/proxy/src/main/java/net/md_5/bungee/UserConnection.java
++++ b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
+@@ -4,6 +4,7 @@ import com.google.common.base.Preconditions;
+ import com.google.common.collect.HashMultimap;
+ import com.google.common.collect.ImmutableMap;
+ import com.google.common.collect.Multimap;
++import io.github.waterfallmc.waterfall.adventure.AdventureTitle; // Waterfall - we're going on an adventure!
```

Comment:
To self: possibly move this down with its friends down below, so they're all in the same block, to avoid confusion.